library/include/BitArray.h

```diff
@@ -528,7 +530,7 @@ namespace DFHack
         {
             auto root = static_cast<L *>(this);
             auto link = new L();
-            link->prev = nullptr;
+            link->prev = root;
             if (root->next)
             {
                 root->next->prev = link;
```

Comment:
Remark: Toady's own `tlist` class _always_ sets `link->next = root->next;`, though if `new L();` initializes both pointers to NULL it should be logically equivalent.

Reply:
I only looked at the constructor for `job_list_link`, and that does indeed null all components. Other than that, I'm not sure what you mean. This is about the question whether the `prev` pointer of the first (real) list element points back to the static `root` node. From my observations, this seems to be the case when DF adds the first element.

Reply:
I wasn't commenting on your fix, just observing another minor difference (which appears to be purely cosmetic) between DFHack's code and Toady's code.